ambry-tools/src/test/java/com.github.ambry/store/BlobIdTransformerTest.java

```diff
@@ -155,8 +155,8 @@ public void testBrokenStoreKeyConverter() throws Exception {
     inputAndExpected = new InputAndExpected(pairList.get(1), VALID_MESSAGE_FORMAT_INPUT_STREAM_IMPLS[0]);
     try {
       transformer.transform(inputAndExpected.getInput());
-      fail("Did not throw MockStoreKeyConverterException");
-    } catch (BlobIdTransformerTestException e) {
+      fail("Did not throw StoreKeyConverterNotConvertedException");
+    } catch (StoreKeyConverter.StoreKeyConverterNotConvertedException e) {
```

Comment:
im not sure we need a new exception type for this - can be just `IllegalStateException` I think.

Reply:
I thought that too at first, but isn't that a runtime exception though, i.e. non-checked?

Reply:
I'm ok with this but if some piece of code is using exceptions for control flow, then perhaps that code or the API is not working very well - so I think `IllegalStateException` is enough here.

@pnarayanan : what do you think?

In any case, if he's ok with the changes, then this can be merged

Reply:
After in-person discussion decided to go with IllegalStateException to avoid encouraging bad control flow paradigms